qcodes/dataset/descriptions/dependencies.py

```diff
@@ -257,6 +269,20 @@ def _to_dict(self) -> InterDependencies_Dict:
                                           "standalones": standalones}
         return output
 
+    def _empty_data_dict(self) -> Dict[str, Dict[str, np.ndarray]]:
```

Comment:
Docstring

Reply:
Considder refactoring the logic that preallocates the cache into prepare too so we can simplify the conditions that cache update must run under

Reply:
Added a prepare method to the cache. Will leave the work of refactoring allocation into this method to a different pr